src/urlMatcherFactory.js

```diff
@@ -122,7 +122,11 @@ function UrlMatcher(pattern, config, parentMatcher) {
     cfg         = config.params[id];
     segment     = pattern.substring(last, m.index);
     regexp      = isSearch ? m[4] : m[4] || (m[1] == '*' ? '.*' : null);
-    type        = $$UMFP.type(regexp || "string") || inherit($$UMFP.type("string"), { pattern: new RegExp(regexp, config.caseInsensitive ? 'i' : undefined) });
```

Comment:
Yes this was the bit that was setting things as type string for me. I didn't fully understand all the code so I tried to confine my changes to `getType` as to limit impact.